src/map/unit.cpp

```diff
@@ -493,6 +532,24 @@ static TIMER_FUNC(unit_walktoxy_timer){
 				clif_move(ud);
 			}
 			break;
+		case BL_NPC:
+			if (nd->sc.option&OPTION_INVISIBLE)
+				break;
+
+			int xs = -1, ys = -1;
+			switch (nd->subtype) {
+			case NPCTYPE_SCRIPT:
+				xs = nd->u.scr.xs;
+				ys = nd->u.scr.ys;
+				break;
+			case NPCTYPE_WARP:
+				xs = nd->u.warp.xs;
+				ys = nd->u.warp.ys;
+				break;
+			}
+			if (xs > -1 || ys > -1)
```

Comment:
This doesn't require both values to be > -1? To assume it's on a valid cell.

Reply:
Hmmm you're right I followed a mistake in map.cpp